.github/workflows/contracts.yml

```diff
@@ -0,0 +1,200 @@
+name: Solidity Random Beacon
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  push:
+    branches:
+      - main
+    paths:
+      - "solidity/random-beacon/**"
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment for workflow execution'
+        required: false
+        default: 'dev'
+      upstream_builds:
+        description: 'Upstream builds'
+        required: false
+      upstream_ref:
+        description: 'Git reference to checkout (e.g. branch name)'
+        required: false
+        default: 'main'
+
+jobs:
+  contracts-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      path-filter: ${{ steps.filter.outputs.path-filter }}
+    steps:
+      - uses: actions/checkout@v2
+        if: github.event_name == 'pull_request'
+
+      - uses: dorny/paths-filter@v2
+        if: github.event_name == 'pull_request'
+        id: filter
+        with:
+          filters: |
+            path-filter:
+              - './solidity/random-beacon/**'
+
+  contracts-lint:
+    needs: contracts-detect-changes
+    if: |
+      github.event_name == 'push'
+        || needs.contracts-detect-changes.outputs.path-filter == 'true'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity/random-beacon
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Lint
+        run: yarn lint
+  
+  contracts-slither:
+    needs: contracts-detect-changes
+    if: |
+      github.event_name == 'push'
+        || needs.contracts-detect-changes.outputs.path-filter == 'true'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity/random-beacon
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          cache-dependency-path: solidity/random-beacon/yarn.lock
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8.5
+
+      - name: Install Solidity
+        env:
+          SOLC_VERSION: 0.8.6 # according to solidity.version in hardhat.config.js
+        run: |
+          pip3 install solc-select
+          solc-select install $SOLC_VERSION
+          solc-select use $SOLC_VERSION
+
+      - name: Install Slither
+        env:
+          SLITHER_VERSION: 0.8.0
+        run: pip3 install slither-analyzer==$SLITHER_VERSION
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Run Slither
+        run: slither --hardhat-artifacts-directory build .
```

Comment:
I see that the command is [the same as in `tbtc-v2` project](https://github.com/keep-network/tbtc-v2/blob/44b78bc6c63caa9736509999501a1c182a3650d5/.github/workflows/contracts.yml#L237) and that in [the workflow for `coverage-pools`](https://github.com/keep-network/coverage-pools/blob/3ac340960dbb3729da3285fa8142cdf3bdaa413e/.github/workflows/contracts.yml#L232) we don't pass the `--hardhat-artifacts-directory` param (instead we specify `hardhat_artifacts_directory` option in [`slither.config.json`](https://github.com/keep-network/coverage-pools/blob/3ac340960dbb3729da3285fa8142cdf3bdaa413e/slither.config.json#L4)). Are both those approaches valid, or is the particular approach dictated by some maters?

Also, I wonder if we don't need to add the `slither.config.json` file with `detectors_to_exclude` and `filter_paths` options? I can see this configuration in both `coverage-pools` and `tbtc-v2`.

Reply:
I think both approaches are fine, it just depends on how many flags we want to pass. If it was just one flag, I guess the inline approach is fine.  But `detectors_to_exclude` also might be handy, so I moved params to `slither.config.json`. Good point.